os/board/rtl8721csm/src/component/os/tizenrt/rtk_netmgr.c

```diff
@@ -436,6 +527,23 @@ trwifi_result_e wifi_netmgr_utils_connect_ap(struct netdev *dev, trwifi_ap_confi
 		}
 	}
 
+	ap_channel = 0xffff;
+
+	if (scan_number) {
+		int i;
+		rtw_mutex_get(&scanlistbusy);
+		for (i = 0 ; i < scan_number ; i++) {
+			if((strncmp(saved_scan_list[i].ssid, ap_connect_config->ssid, ap_connect_config->ssid_length) == 0)

```

Comment:
if scan_number is not protected then this situation might be happened.

1. wifi_manager requests scan.
2. a driver calls save_scan_list and set a timer
3. after 3minutes later, a timer is expired. so the timer execute rtw_mfree in _scan_timer_handler().
4. before assign scan_number = 0 in _scan_timer_handler() context switching happen.
5. wifi_manager request connect. and go inside if(scan_number) because scan_number is not 0.
6. connec function would access invalid memory (saved_scan_list[i]).

Reply:
@pillip8282, Modified to protect three shared variables of latest added code with `_mutex scanlistbusy`